lib/vistle/control/hub.cpp

```diff
@@ -1882,6 +1955,36 @@ void Hub::clearStatus()
     m_statusText.clear();
 }
 
+int Hub::getParentCompound(int modId)
+{
+    if (!Id::isModule(modId))
+        return Id::Invalid;
+    auto modName = m_stateTracker.getModuleName(modId);
+    for (size_t i = modId - 1; i >= message::Id::ModuleBase; --i) {
+        auto modit = m_stateTracker.runningMap.find(i);
+        if (modit != m_stateTracker.runningMap.end()) {
+            auto &av = m_stateTracker.getStaticModuleInfo(modit->first);
+        if (av.isCompound()) {
+                if (av.submodules().size() >= modId - i && av.submodules()[modId - i - 1].name == modName)
```

Comment:
what if a compound contains two or more instance of the same module?

Reply:
av.submodules().size() >= modId - sould always be a sufficent check. av.submodules()[modId - i - 1].name == modName ist just to be extra save.

Reply:
make the last check an assert() then?